pytest_allclose/tests/test_allclose.py

```diff
@@ -0,0 +1,106 @@
+"""Test the allclose fixture."""
+
+import numpy as np
+import pytest
+
+
+def eye_vector(n, k, dtype=bool):
+    return np.eye(1, n, k=k, dtype=dtype)[0]
+
+
+def add_close_noise(x, atol, rtol, rng):
+    m = rng.uniform(-rtol, rtol, size=x.shape)
+    n = rng.uniform(-atol, atol, size=x.shape)
+    return x + m*np.abs(x) + n
+
+
+def get_vector_pairs(atol, rtol, rng):
+    x = rng.uniform(-1, 1, size=100)
+    y = add_close_noise(x, atol=atol, rtol=rtol, rng=rng)
+
+    m = eye_vector(x.size, 0)
+    x0 = x * (~m)
+    y0 = x0 + 1.1*atol*m
+
+    i = np.argmax(np.abs(x))
+    assert np.abs(x[i])*rtol > atol
+    x1 = x
+    y1 = x1 + 3*rtol*x1[i]*eye_vector(x.size, i)
+
+    return [
+        (True, x, y),
+        (False, x0, y0),
+        (False, x1, y1),
+    ]
+
+
+def test_allclose(allclose):
+    atol = 1e-5
+    rtol = 1e-3
+
+    pairs = get_vector_pairs(atol, rtol, np.random.RandomState(3))
+    for close, x, y in pairs:
+        assert allclose(y, x, atol=atol, rtol=rtol) == close
```

Comment:
Because of the way `allclose` works. `allclose(a, b)` returns true if `abs(a - b) <= rtol*abs(b) + atol`. i.e. `b` is used to determine the relative error tolerance. Since `x` is always our reference signal that we're comparing to, it makes more sense to me to take the relative tolerance with respect to it. In tests, `x` will generally be consistent (i.e. it's what the output should be), and so the error tolerance will be consistent. So it's just become a habit for me that I do it that way.

Reply:
Interesting. I've never treated the two arguments as different before -- which is probably why I generally ignore the `rtol` value and just play with the `atol`. I wish NumPy would have used better names than `a` and `b` since they are functionally different (e.g. `test` and `ref` or something like that) but whatcha gonna do.